tracer/tracer.go

```diff
@@ -155,6 +176,55 @@ func (tracer *epsagonTracer) addRunnerException(event *protocol.Event) {
 	}
 }
 
+func isStrongKey(key string) bool {
+	_, ok := strongKeys[key]
+	return ok
+}
+
+func (tracer *epsagonTracer) stripEvents(traceLength int, marshaler *jsonpb.Marshaler) {
+	eventSize := 0
+	for _, event := range tracer.events {
+		eventJSON, err := marshaler.MarshalToString(event)
+		if err != nil {
+			continue
+		}
+		eventSize = len(eventJSON)
+		for key, _ := range event.Resource.Metadata {
+			if !isStrongKey(key) {
+				delete(event.Resource.Metadata, key)
+			}
+		}
+		eventJSON, err = marshaler.MarshalToString(event)
+		if err != nil {
+			continue
+		}
+		strippedSize := eventSize - len(eventJSON)
+		traceLength -= strippedSize
+		if traceLength <= MaxTraceSize {
+			return
+		}
+	}
+}
+
+func (tracer *epsagonTracer) getTraceJSON(trace *protocol.Trace, runnerEvent *protocol.Event) (traceJSON string, err error) {
+	marshaler := jsonpb.Marshaler{
+		EnumsAsInts: true, EmitDefaults: true, OrigName: true}
+	traceJSON, err = marshaler.MarshalToString(trace)
+	if err != nil {
+		return
+	}
+	traceLength := len(traceJSON)
+	if traceLength > MaxTraceSize {
+		tracer.stripEvents(traceLength, &marshaler)
+		runnerEvent.Resource.Metadata[IsTrimmedKey] = "true"
+		if tracer.Config.Debug {
+			log.Printf("EPSAGON DEBUG trimmed trace (max allowed size: 64KB)")
```

Comment:
I would move this print inside the stripEvents function, and print the original size and trimmed size as well. What do you think?